### ISSUE TITLE
 java.lang.NoClassDefFoundError: scala/reflect/api/TypeCreator Crash

### ISSUE THREAD

**Issue by ronyhe** [com-lihaoyi/PPrint#24]
When trying to use the library I get the following error:
```text
[error] (run-main-0) java.lang.NoClassDefFoundError: scala/reflect/api/TypeCreator
[error] java.lang.NoClassDefFoundError: scala/reflect/api/TypeCreator
[error] 	at fansi.ColorCategory.$anonfun$Full$1(Fansi.scala:851)
[error] 	at fansi.ColorCategory.$anonfun$Full$1$adapted(Fansi.scala:850)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
[error] 	at scala.collection.immutable.Range.foreach(Range.scala:158)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:237)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:230)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at fansi.ColorCategory.<init>(Fansi.scala:850)
[error] 	at fansi.Color$.<init>(Fansi.scala:717)
[error] 	at fansi.Color$.<clinit>(Fansi.scala)
[error] 	at pprint.PPrinter$.$lessinit$greater$default$4(PPrinter.scala:18)
[error] 	at pprint.package$.<init>(package.scala:7)
[error] 	at pprint.package$.<clinit>(package.scala)
[error] 	at com.ronyhe.minerva.Main$.printParse(Main.scala:17)
[error] 	at com.ronyhe.minerva.Main$.main(Main.scala:9)
[error] 	at com.ronyhe.minerva.Main.main(Main.scala)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] Caused by: java.lang.ClassNotFoundException: scala.reflect.api.TypeCreator
[error] 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:588)
[error] 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
[error] 	at fansi.ColorCategory.$anonfun$Full$1(Fansi.scala:851)
[error] 	at fansi.ColorCategory.$anonfun$Full$1$adapted(Fansi.scala:850)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
[error] 	at scala.collection.immutable.Range.foreach(Range.scala:158)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:237)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:230)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at fansi.ColorCategory.<init>(Fansi.scala:850)
[error] 	at fansi.Color$.<init>(Fansi.scala:717)
[error] 	at fansi.Color$.<clinit>(Fansi.scala)
[error] 	at pprint.PPrinter$.$lessinit$greater$default$4(PPrinter.scala:18)
[error] 	at pprint.package$.<init>(package.scala:7)
[error] 	at pprint.package$.<clinit>(package.scala)
// These are my methods
[error] 	at com.ronyhe.minerva.Main$.printParse(Main.scala:17)
[error] 	at com.ronyhe.minerva.Main$.main(Main.scala:9)
[error] 	at com.ronyhe.minerva.Main.main(Main.scala)
// End of my methods
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error] Nonzero exit code: 1

Any ideas?
```

**Comment by ronyhe** [comment]
This is with scalaVersion := "2.12.8"

**Comment by pedrotst** [comment]
Confirm same problem with version 2.12.3

**Comment by halaharvi** [comment]
Looks like SBT cannot find the scala-reflect package. 

Did you try adding the `scala-reflect.jar` manually to your `build.sbt`? 

`"org.scala-lang" % "scala-reflect" % "2.12.8"`

**Comment by diyan** [comment]
@halaharvi Thank you. Yours hint works for me.

```scala
// build.sbt:
// "com.lihaoyi" %% "pprint" % "0.5.3",
// "org.scala-lang" % "scala-reflect" % "2.12.8",
import pprint.pprintln
case class User(name: String, age: Option[Int])
val users = List(User("John", Some(30)), User("Anna", None))
pprintln(users)
// OUTPUT:
// List(User("John", Some(30)), User("Anna", None))
```

**Comment by imranbutt-java** [comment]
It happens, when we have added `% "provided"` in `build.sbt`
After removing `% "provided"`, it resolved the problem for me.
But for sure for sbt assembly I have to put them back to exclude unnecessary jars to be the part of my package.

**Comment by lihaoyi** [comment]
Seems to have been fixed somewhere between pprint 0.5.3 and pprint 0.6.6, closing
